components/org.wso2.carbon.event.outbound.adapter.websubhub/src/main/java/org/wso2/carbon/event/outbound/adapter/websubhub/util/WebSubHubEventAdapterUtil.java

```diff
@@ -0,0 +1,320 @@
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Publishing event data to WebSubHub. URL: " + url + " tenant domain: "
+                    + tenantDomain);
+        }
+
+        client.execute(request, new FutureCallback<HttpResponse>() {
+            @Override
+            public void completed(final HttpResponse response) {
+
+                int responseCode = response.getStatusLine().getStatusCode();
+                if (log.isDebugEnabled()) {
+                    log.debug("WebSubHub request completed. Response code: " + responseCode);
+                }
+                if (responseCode == 200 || responseCode == 201 || responseCode == 202 || responseCode == 204) {
+                    // Check for 200 success code range.
+                    if (log.isDebugEnabled()) {
+                        String jsonString;
+                        try {
+                            jsonString = EntityUtils.toString(response.getEntity());
+                            JSONParser parser = new JSONParser();
+                            JSONObject json = (JSONObject) parser.parse(jsonString);
+                            log.debug("Response data: " + json);
+                        } catch (IOException | ParseException e) {
+                            log.debug("Error while reading WebSubHub event publisher response. ", e);
+                        }
+                    }
+                } else {
+                    log.error("WebHubSub event publisher received " + responseCode + " code.");
+                    String jsonString;
+                    try {
+                        jsonString = EntityUtils.toString(response.getEntity());
+                        JSONParser parser = new JSONParser();
+                        JSONObject json = (JSONObject) parser.parse(jsonString);
+                        log.error("Response data: " + json);
+                    } catch (IOException | ParseException e) {
+                        log.error("Error while reading WebSubHub event publisher response. ", e);
+                    }
+                }
+            }
+
+            @Override
+            public void failed(final Exception ex) {
+
+                log.error("Publishing event data to WebSubHub failed. ", ex);
+            }
+
+            @Override
+            public void cancelled() {
+
+                log.error("Publishing event data to WebSubHub cancelled.");
+            }
+        });
+
+    }
+
+    public static void makeTopicMgtAPICall(String topic, String webSubHubBaseUrl, String operation)
+            throws IOException, WebSubAdapterServerException {
+
+        String topicMgtUrl = buildURL(topic, webSubHubBaseUrl, operation);
+
+        try (CloseableHttpClient httpClient = HttpClientBuilder.create().useSystemProperties().build()) {
+
+            HttpPost httpPost = new HttpPost(topicMgtUrl);
+            httpPost.setHeader(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType());
```

Comment:
Shall we include correlation-id to the request headers as same way we have done in event publishing requests?

`httpPost.setHeader(CORRELATION_ID_REQUEST_HEADER, getCorrelationID());`